### PR TITLE
Adding builder image and documentation

### LINKED ISSUE #13: Include builder image
The Contributing guide has [instructions on building the site with Docker](https://github.com/redhat-cop/uncontained.io/blob/master/CONTRIBUTING.md#containerized-hugo-environment-quickstart-guide), but no Hugo builder is included.

Let's create a builder image for Hugo, or find and include an existing open source tool that does the same thing. This builder image should include Hugo, Ruby, and Asciidoctor. Here's [one example](https://github.com/blerchin/hugo-webpack-boilerplate/blob/master/Dockerfile).

### PR DESCRIPTION
**Pull request by etsauer**
#### What is this PR About?
Adds a builder image for testing changes to the site.

#### How should we test or review this PR?

See new usage instructions in:

- CONTRIBUTING.md
- container-image/hugo/README.adoc

#### Is there a relevant Trello card or Github issue open for this?
resolve #13 

#### Who would you like to review this?
cc: @redhat-cop/cant-contain-this

-----

- [x] Have you followed the [contributing guidelines](https://github.com/redhat-cop/uncontained.io/blob/master/CONTRIBUTING.md)?
- [x] Have you explained what your changes do, and why they add value to the uncontained.io guides?

-----

**Please note: we may close your PR without comment if you do not check the boxes above and provide ALL requested information.**

### CODE REVIEW

File: container-images/hugo/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+ENV HOME /opt/hugo
+ENV USER 1001
+
+USER root
+
+RUN curl -sLo /tmp/hugo.tar.gz ${HUGO_URL} \
+  && tar zxf /tmp/hugo.tar.gz \
+  && mv hugo /bin/hugo
+
+ADD ./container-images/hugo/root /
```
Comment (JaredBurck) [inline container-images/hugo/Dockerfile]: docker build fails here
```
Step 10/15 : ADD ./container-images/hugo/root /
ADD failed: stat /var/lib/docker/tmp/docker-builder155454769/container-images/hugo/root: no such file or directory
```

**Comment by etsauer** [comment]
@JaredBurck fixed. my `.gitignore` was too powerful... left out some things i need.

File: container-images/hugo/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,26 @@
+FROM centos/ruby-24-centos7
```
Comment (JaredBurck) [inline container-images/hugo/Dockerfile]: I have a builder image that I have been working on creating a builder image based on alpine. It was inspired by the Hugo and what they use to generate their website [1].  You can see my dockerfile in my branch [2].  Any thoughts about going in this direction?

[1] https://github.com/jojomi/docker-hugo
[2] https://github.com/JaredBurck/uncontained.io/blob/hugo-builder-image/containers/hugo-builder/Dockerfile

Comment (JaredBurck) [reply container-images/hugo/Dockerfile]: will merge and continue conversation on idea as a follow

**Pull request merged**